### PR TITLE
Add `@oneOf` support to introspection query

### LINKED ISSUE #4077: Introspection queries don't support `@oneOf`
Related to the issue in https://github.com/graphql/graphql-js/issues/3968, it doesn't appear that the introspection query is selecting the `isOneOf` field from the server, as defined by the [RFC](https://github.com/graphql/graphql-spec/pull/825/files#diff-30a69c5a5eded8e1aea52e53dad1181e6ec8f549ca2c50570b035153e2de1c43).

This has downstream effects like preventing this introspection response from powering codegen that uses that flag to enforce mutual exclusivity amongst the arguments.

I've attempted to add this functionality in https://github.com/graphql/graphql-js/pull/4078

### PR DESCRIPTION
**Pull request by maciesielka**
Fixes #4077

### CODE REVIEW

File: src/utilities/getIntrospectionQuery.ts [hunk 1]
```diff
@@ -32,6 +32,10 @@ export interface IntrospectionOptions {
    * Default: false
    */
   inputValueDeprecation?: boolean;
+
+  // Whether target GraphQL server supports `@oneOf` input objects.
+  // Default: false
+  inputObjectOneOf?: boolean;
```
Comment (maciesielka) [inline src/utilities/getIntrospectionQuery.ts:38]: Followed the pattern here above for making this configurable, but wasn't sure if that was necessary.

**Review by JoviDeCroock — APPROVED** [review]

File: src/utilities/getIntrospectionQuery.ts [hunk 2]
```diff
@@ -32,6 +32,10 @@ export interface IntrospectionOptions {
    * Default: false
    */
   inputValueDeprecation?: boolean;
+
+  // Whether target GraphQL server supports `@oneOf` input objects.
+  // Default: false
```
Comment (mmgoodnow) [inline src/utilities/getIntrospectionQuery.ts:37]: should this be JSDoc formatted to match the comments above?

File: src/utilities/getIntrospectionQuery.ts [hunk 3]
```diff
@@ -223,6 +230,7 @@ export interface IntrospectionObjectType {
   readonly interfaces: ReadonlyArray<
     IntrospectionNamedTypeRef<IntrospectionInterfaceType>
   >;
+  readonly isOneOf: boolean;
```
Comment (mmgoodnow) [inline src/utilities/getIntrospectionQuery.ts:233]: looks like this was added to the `IntrospectionObjectType` here in addition to the `IntrospectionInputObjectType` below—is that intentional? The other references in this diff seem to only care about input objects.

File: src/utilities/getIntrospectionQuery.ts (hunk 2 above)
Comment (maciesielka) [reply src/utilities/getIntrospectionQuery.ts]: 🤦🏻 definitely. updated in [3af4434](https://github.com/graphql/graphql-js/pull/4078/commits/3af443467cdb9713f700e2b7b4368fb55c18171c)

File: src/utilities/getIntrospectionQuery.ts (hunk 3 above)
Comment (maciesielka) [reply src/utilities/getIntrospectionQuery.ts]: You're so right. Removed in [633331a](https://github.com/graphql/graphql-js/pull/4078/commits/633331aded549b3db00296646ad4833b5f8b5c7f)

**Comment by maciesielka** [comment]
@JoviDeCroock, @mmgoodnow appreciate your reviews and hopefully I've addressed your feedback. Any idea what it takes to get something like this merged?

**Review by benjie — APPROVED** [review]
Thanks for adding this vital feature!

**Pull request merged**